File: gpfksdlrn/level1/폰켓몬.js
// https://school.programmers.co.kr/learn/courses/30/lessons/1845

function solution(nums) {
  return Math.min(
    nums.length / 2, // 가져갈 수 있는 마리 수
    new Set(nums).size, // 폰켓몬 종류 수
  );
}

File: gnsals0904/level1/폰켓몬.java
class Solution {
    static int[] hashSort = new int[200001];
    public int solution(int[] nums) {
        for(int i : nums){
            hashSort[i]++;
        }
        int answer = 0;
        for(int i = 0; i < 200001; i++){
            if(hashSort[i] > 0){
                answer++;
            }
        }
        if(answer > nums.length/2) return nums.length/2;
        return answer;
    }
}